clinvar-variant-types/clinvar-variant-types.py

```diff
@@ -40,116 +89,172 @@ def find_attribute(rcv, xpath, attribute_name):
+    'Distribuion of records by star rating', 'Star rating', sort_lambda=lambda x: x[0])
+table_multiple_mode_of_inheritance = SupplementaryTable(
+    'Multiple mode of inheritance', ['RCV', 'Modes of inheritance'], sort_lambda=lambda x: (x[1], x[0]))
+counter_multiple_allele_origin = SupplementaryTableCounter('Multiple allele origins', 'Allele origins')
+table_inconsistent_moi_ao = SupplementaryTable(
+    'Inconsistent mode of inheritance and allele origin values', ['RCV', 'Modes of inheritance', 'Allele origins'],
+    sort_lambda=lambda x: (x[1], x[2], x[0]))
 
 elements_processed = 0
-for rcv in clinvar_xml_utils.iterate_rcv_from_xml(args.clinvar_xml):
-    # TODO: use clinvar_xml_utils abstractions instead of working with XML directly
-    # TODO: https://github.com/EBIvariation/eva-opentargets/issues/170
-    rcv_id = rcv.find('./ClinVarAccession').attrib['Acc']
+for clinvar_record in clinvar_xml_utils.ClinVarDataset(args.clinvar_xml):
+    rcv_id = clinvar_record.accession
 
     # RCV can contain either a MeasureSet, or a GenotypeSet. It must not contain both.
-    measure_sets = rcv.findall('MeasureSet')
-    genotype_sets = rcv.findall('GenotypeSet')
+    measure_sets = clinvar_record.rcv.findall('MeasureSet')
+    genotype_sets = clinvar_record.rcv.findall('GenotypeSet')
     if len(measure_sets) == 1 and len(genotype_sets) == 0:
         # Most common case. RCV directly contains one measure set.
         measure_set = measure_sets[0]
         measure_set_type = measure_set.attrib['Type']
-        add_transitions(variant_type_transitions, ('RCV', 'MeasureSet', measure_set_type))
+        sankey_variation_representation.add_transitions('RCV', 'MeasureSet', measure_set_type)
 
         if measure_set_type == 'Variant':
-            # Most common case, accounting for >99.95% of all ClinVar records.. Here, we go into details on various
+            # Most common case, accounting for >99.97% of all ClinVar records. Here, we go into details on various
             # attribute distributions.
 
-            # Variant type
+            # Variation representation
             measures = measure_set.findall('Measure')
             assert len(measures) == 1, 'MeasureSet of type Variant must contain exactly one Measure'
-            add_transitions(variant_type_transitions, (measure_set_type, measures[0].attrib['Type']))
+            sankey_variation_representation.add_transitions(measure_set_type, measures[0].attrib['Type'])
+
+            # Trait representation
+            traits = clinvar_record.traits
+            if len(traits) == 0:
+                traits_category = 'No traits'
+            elif len(traits) == 1:
+                traits_category = 'One trait'
+            else:
+                traits_category = 'Multiple traits'
+            names_category = 'One name per trait'
+            for trait in traits:
+                if len(trait.all_names) > 1:
+                    names_category = 'Multiple names per trait'
```

Comment:
This will mean that the `'No traits'` ends up with `'One name per trait'` where I think you should stop the sankey diagram

Reply:
First, you are right and that was a mistake. But second, that case only ever appeared due to the fact that pipeline versions 2.0.0 ... 2.0.1 only processed `Type="Disease"` traits. This is now fixed, and turns out there isn't a single case with “No traits” (which actually makes sense, otherwise a ClinVar record would just be useless)